src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -372,6 +372,43 @@ private bool TryReplaceBannedFunction(FunctionExpression expression, [NotNullWhe
                 return true;
             }
 
+            if (StringComparer.OrdinalIgnoreCase.Equals(expression.Function, "lambda"))
+            {
+                if (expression.Parameters.Length < 1)
+                {
+                    throw new ArgumentException($"Expected at least 1 argument for function {expression.Function}");
+                }
+
+                var lambdaExpression = expression.Parameters.Last();
+                var paramNames = new string[expression.Parameters.Length - 1];
+                for (var i = 0; i < expression.Parameters.Length - 1; i++)
+                {
+                    if (expression.Parameters[i] is not JTokenExpression jtokenParam ||
+                        jtokenParam.Value.Type is not JTokenType.String)
+                    {
+                        throw new ArgumentException($"Argument {i} for {expression.Function} is not a valid identifier");
+                    }
+
+                    paramNames[i] = jtokenParam.Value.ToString();
+                }
+
+                syntax = SyntaxFactory.CreateLambdaSyntax(paramNames, ParseLanguageExpression(lambdaExpression));
+                return true;
+            }
+
+            if (StringComparer.OrdinalIgnoreCase.Equals(expression.Function, "lambdaVariables"))
+            {
+                if (expression.Parameters.Length != 1 ||
+                    expression.Parameters[0] is not JTokenExpression jtokenParam ||
+                    jtokenParam.Value.Type is not JTokenType.String)
+                {
+                    throw new ArgumentException($"Expected exactly 1 parameter of type {JTokenType.String} for function {expression.Function}");
```

Comment:
Why does the parameter have to be of type `JTokenType.String`?

Reply:
Lambdas are represented in ARM with the `lambda` function, with lambda variable accesses represented with the `lambdaVariables` function.

For example, the following lambda:
`(a, b) => a + b`
Would be represented as:
`lambda('a', 'b', add(lambdaVariables('a'), lambdaVariables('b'))`

So it doesn't make sense (and wouldn't work) for the value provided to the `lambdaVariables` function to be anything other than the name of the variable ('a' or 'b' in this example) to be accessed.

Reply:
Ah makes sense! Thanks